### ISSUE TITLE
[REQUEST] Filter: CatchOnlyAboveIv

### ISSUE THREAD

**Issue by Galactimus** [NoxxDev/NecroBot#823]
Hello there, I've seen lots of people getting banned cause of sniping too many rare pokemons like dragonite, lapras and snorlax These guys gather like 900 candies due to crazy sniping looking for a 100iv monster, so they end up to have a huge quantity of dragonite, lapras and snorlax catched and seen which is  pretty impossible even for the most able and high level trainers. You know all catched and seen pokemon informations are stored on Niantic server in each trainer log, you can see this log from your app yourself too, however Niantic can analyze these informations, like sum up the number of dragonite you catched, or simply check the number of rare candies you have for snorlax and lapras, and if this number is kinda suspect, niantic will of course ban you. That's why I'm really suggesting you to implement a filter like the one i wrote as title of this request, it would be better if you can implement this filter as selective for each pokemon, for example I want to catch only 100iv snorlax so I set this filter only for it. By using such a filter you will not get noticed by niantic cause you will catch only a few "rare" pokemon, but they will be the 100iv monsters you were looking for. It should be pretty easy to implement comparing to other complex feauters, in fact you just have to modify the "CatchPokemonTask.cs" by adding for example some lines after the iv and cp calculation, like that:

//Calculate CP and IV
            var pokemonCp = (encounter is EncounterResponse
                               ? encounter.WildPokemon?.PokemonData?.Cp
                               : encounter.PokemonData?.Cp);
            var pokemonIv = PokemonInfo.CalculatePokemonPerfection(encounter is EncounterResponse
                    ? encounter.WildPokemon?.PokemonData
                    : encounter?.PokemonData);
//CatchOnlyAboveIV FILTER
            var idChecker=encounter is EncounterResponse ? pokemon.PokemonId : encounter?.PokemonData.PokemonId;
            if(idChecker==131&&pokemonIv<100) return;
            else if(idChecker==143&&pokemonIv<100) return;
            else  if(idChecker==149&&pokemonIv<100) return;

However by using this filter you will still have a huge amount of rare (like dragonite) pokemon seen but I don't think niantic is taking it into account right now. In order to overcome this issue too there should be a way to determine a pokemon IV without entering in a battle with it, but I sincerely think something like that is not possible, if you know this can be done you are welcome! By the way please implement a filter like the one I'm writing this post for as soon as possible

**Comment by iwean** [comment]
And after walk and walk with never ending and catch even 1 pokemon.... I get this :smile: 
....
![screenshot_171](https://cloud.githubusercontent.com/assets/4365551/18223701/f6527d00-71e9-11e6-807e-06fa06612945.jpg)

**Comment by Galactimus** [comment]
Dude I just wrote some code as example for the developers, I've never said it is correct cause I don't neither know how to compile the source code properly, if you wanna explain it to me you are welcome, however just try to do an explicit cast like that and let me know if it works:

//CatchOnlyAboveIV FILTER
var idChecker=encounter is EncounterResponse ? pokemon.PokemonId : encounter?.PokemonData.PokemonId;
if((int)idChecker==131&&(int)pokemonIv<100) return;
else if((int)idChecker==143&&(int)pokemonIv<100) return;
else if(int)idChecker==149&&(int)pokemonIv<100) return;

**Comment by iwean** [comment]
Yeah... work for a couple of minutes, but make the bot just walk over & over pokestop to spin but not for catching a single pokemon.... than it crash like on my screenshot. It seem had a conflict with another task when it wake for it sequence.

But I'm agree with your argument that crazy sniping will make Niantic trace you. Anyone can implement this filter?

**Comment by Silph-Road** [comment]
@Galactimus @iwean @NoxxDev @jjskuld @mo0ojava 

I too agree with this request.
Niancitc can definitely track that you are only catching rares and have a ton of candy.
I think that the ban wave are also based on how many snorlax, lapras and other non evolution candy one has.


It would be helpful to make to a setting for each pokemon like:

"Snorlax": {
      **"CatchMinLvl": 20,
      "CatchMinCP": 2000,
      "CatchMinIV": 90,
      "UseCatchMin": true,
      "CatchMinOperator": "or",**
      "KeepMinCp": 10,
      "KeepMinLvl": 22,
      "UseKeepMinLvl": true,
      "KeepMinIvPercentage": 90.0,
      "KeepMinDuplicatePokemon": 40,
      "Moves": [],
      "DeprecatedMoves": [],
      "KeepMinOperator": "or",
      "MovesOperator": "or"
    },

A global setting would not allow the bot to catch anything else and that would waste a lot of bot time. I think I can catch maybe 1-2 100% IV per day, but i would like to be able to catch other pokemons to be able to level up and get candy eg, dratini candy eevee candy etc.

So a setting specific for each rare pokemon snorlax, lapras, scyther etc will help.


What do you guys think? Do you like the specific setting per pokemon or a global one?

**Comment by iwean** [comment]
Spesific setting of cource... :+1: 
Because if it for global..... you just walk & walk and only catch 1 - 2 with 100% IV per day.

**Comment by mo0ojava** [comment]
You can do a rudimentary version of this already. Set min IV in Pogofeeder to desired value (95?). Then restrict types by using the filter and also check the  "catchonlypokeswithknownIV"-box (don't remember the exact name but it's in there).

**Comment by Silph-Road** [comment]
true, but regular teleport snipe too risky, i prefer human walking.

@mo0ojava 
But i have a question regarding using pogo feeder.
does it work in a way that is described here (causing bans):
[http://pokemongo.nianticlabs.com/en/post/update-082916/](http://pokemongo.nianticlabs.com/en/post/update-082916/)

**Comment by Silph-Road** [comment]
@NoxxDev 
@mo0ojava 
@jjskuld 
Off topic, but I came across this article.
It talks about what information niantic uses to detect bot players.

> Niantic general counsel Courtney Greene Power wrote a letter to Sen. Franken (PDF) sying, “Country is collected and stored, to provide a user the appropriate experience; language may be stored in future updates, for the same purpose,The app collects certain information to facilitate important quality and stability objectives and to prevent abuse. This includes information such as mobile operating system, mobile device identifier, and hardware build information. This information is used to debug phone-specific game problems and to detect and avoid cheating in the game.”

[https://otakukart.com/animeblog/2016/09/02/niantic-responds-to-pokemon-go-privacy/](https://otakukart.com/animeblog/2016/09/02/niantic-responds-to-pokemon-go-privacy/)

**Comment by Galactimus** [comment]
Yep I agree, specific setting of course, a global one would be pretty useless
